src/Gaffer/PlugAlgo.cpp

```diff
@@ -0,0 +1,81 @@
+#include "Gaffer/PlugAlgo.h"
+
+using namespace std;
+using namespace IECore;
+
+void replacePlugWalk( Plug *existingPlug, Plug *plug, ConnectionsVector &connections )
+{
+	// Record output connections.
+	Connections c;
+	c.plug = plug;
+	c.outputs.insert( c.outputs.begin(), existingPlug->outputs().begin(), existingPlug->outputs().end() );
+
+	if( plug->children().size() )
+	{
+		// Recurse
+		for( PlugIterator it( plug ); !it.done(); ++it )
+		{
+			if( Plug *existingChildPlug = existingPlug->getChild<Plug>( (*it)->getName() ) )
+			{
+				replacePlugWalk( existingChildPlug, it->get(), connections );
+			}
+		}
+	}
+	else
+	{
+		// At a leaf - record input connection and transfer values if
+		// necessary. We only store inputs for leaves because automatic
+		// connection tracking will take care of connecting the parent
+		// levels when all children are connected.
+		c.input = existingPlug->getInput<Plug>();
+		if( !c.input && plug->direction() == Plug::In )
+		{
+			ValuePlug *existingValuePlug = runTimeCast<ValuePlug>( existingPlug );
+			ValuePlug *valuePlug = runTimeCast<ValuePlug>( plug );
+			if( existingValuePlug && valuePlug )
+			{
+				valuePlug->setFrom( existingValuePlug );
+			}
+		}
+	}
+
+	connections.push_back( c );
+}
+
+void replacePlug( Gaffer::GraphComponent *parent, PlugPtr plug )
```

Comment:
Ah! I wasn't sure what your "What do I need to do to get this symbol?" comment was about, but having tried to compile your branch I presume it was a linking error where `replacePlug()` wasn't found? You'll either need to declare this here explicitly as `void PlugAlgo::replacePlug( Gaffer::GraphComponent *parent, PlugPtr plug )`, or scope it inside `namespace PlugAlgo {}`. Otherwise the compiler decides to generate a `replacePlug()` symbol in the global namespace rather than line this up with previously declared function in `namespace PlugAlgo`. You can see examples of each approach in SceneAlgo.cpp and StringAlgo.cpp respectively.